config/impl.go

```diff
@@ -21,54 +21,85 @@
 package config
 
 import (
+	"context"
 	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+
 	"go.ketch.com/lib/orlop/v2/env"
 	"go.ketch.com/lib/orlop/v2/errors"
 	"go.ketch.com/lib/orlop/v2/service"
-	"go.uber.org/fx"
-	"reflect"
-	"strings"
 )
 
-type Config interface {
-	Options() fx.Option
-}
-
 type providerImpl struct {
+	configs map[string]interface{}
 	environ env.Environ
 }
 
 func New(environ env.Environ) Provider {
 	return &providerImpl{
+		configs: make(map[string]interface{}, 0),
 		environ: environ,
 	}
 }
 
-func (s *providerImpl) Load(cfg Config) error {
-	fields, err := reflectStruct([]string{}, cfg)
-	if err != nil {
-		return err
+func (s *providerImpl) Register(_ context.Context, service string, cfg interface{}) {
```

Comment:
I am now thinking perhaps we should switch to a more declarative interface, such that we can do the following:
```
fx.Supply(config.Config("server", &Config)),
```

Then rely on an optional group of these `config.ConfigDef` structs when creating this provider.